lib/tasks/validate_content_against_schema.rake

```diff
@@ -0,0 +1,56 @@
+task validate_content_against_schema: :environment do
+  GovukContentSchemaTestHelpers.configure do |config|
+    config.schema_type = 'frontend'
+    config.project_root = Rails.root
+  end
+
+  schema_names = ContentItem.distinct(:schema_name).select(&:present?)
+  puts "Will check #{schema_names}"
+
+  reports = schema_names.map do |schema_name|
+    validate_content_for_schema(schema_name)
+  end
+
+  total_checked = reports.reduce(0) { |s, r| s + r[:counts][:checked] }
+  total_fail = reports.reduce(0) { |s, r| s + r[:counts][:fail] }
+  failure_percentage = ((total_fail / total_checked.to_f) * 100).to_i
+
+  Rails.application.statsd.gauge("document_validation.checked", total_checked)
+  Rails.application.statsd.gauge("document_validation.fail", total_fail)
+  Rails.application.statsd.gauge("document_validation.failure_percentage", failure_percentage)
+
+  puts reports.to_yaml
+end
+
+def validate_content_for_schema(schema_name)
+  api_url_callable = lambda { |base_path| "http://api.example.com/content#{base_path}" }
+
+  # content_id ASC makes the sample semi-random
+  items = ContentItem.order('content_id ASC').where(schema_name: schema_name).limit(10)
+
+  counts = { checked: 0, okay: 0, fail: 0 }
+  errors = []
+
+  items.each do |content_item|
+    print '.'
+
+    counts[:checked] = counts[:checked] + 1
```

Comment:
Should be possible to do `counts[:checked] += 1` and for the various times below.